src/components/popover-container/popover-container.pw.tsx

```diff
@@ -0,0 +1,683 @@
+      mount,
+      page,
+    }) => {
+      await mount(
+        <Box display="flex" justifyContent="center">
+          <PopoverContainerComponent
+            position={position}
+            renderOpenComponent={({
+              onClick,
+            }: {
+              onClick: (ev: React.MouseEvent<HTMLElement>) => void;
+            }) => <Portrait onClick={onClick} />}
+          />
+        </Box>
+      );
+      const popoverContainerContentElement = popoverContainerContent(page);
+
+      await expect(popoverContainerContentElement).toHaveCSS(
+        "position",
+        "absolute"
+      );
+
+      await assertCssValueIsApproximately(
+        popoverContainerContentElement,
+        "top",
+        top
+      );
+
+      await assertCssValueIsApproximately(
+        popoverContainerContentElement,
+        "right",
+        right
+      );
+
+      await assertCssValueIsApproximately(
+        popoverContainerContentElement,
+        "bottom",
+        bottom
+      );
+
+      await assertCssValueIsApproximately(
+        popoverContainerContentElement,
+        "left",
+        left
+      );
+    });
+  });
+
+  test("should render with ariaDescribedBy", async ({ mount, page }) => {
+    await mount(<PopoverContainerComponent ariaDescribedBy={testString} />);
+
+    await expect(popoverContainerContent(page)).toHaveAttribute(
+      "aria-describedby",
+      testString
+    );
+  });
+
+  test("should render with openButtonAriaLabel", async ({ mount, page }) => {
+    await mount(<PopoverContainerComponent openButtonAriaLabel={testString} />);
+
+    const popoverContainerOpenComponent = getDataElementByValue(
+      page,
+      "popover-container-open-component"
+    );
+
+    await expect(popoverContainerOpenComponent).toHaveAttribute(
+      "aria-label",
+      testString
+    );
+  });
+
+  test("should render with closeButtonAriaLabel", async ({ mount, page }) => {
+    await mount(
+      <PopoverContainerComponent closeButtonAriaLabel={testString} />
+    );
+
+    const popoverCloseIconElement = getDataElementByValue(
+      page,
+      "popover-container-close-component"
+    );
+
+    await expect(popoverCloseIconElement).toHaveAttribute(
+      "aria-label",
+      testString
+    );
+  });
+
+  test("should render with containerAriaLabel", async ({ mount, page }) => {
+    await mount(<PopoverContainerComponent containerAriaLabel={testString} />);
+
+    const popoverContainerContentElement = popoverContainerContent(page);
+
+    await expect(popoverContainerContentElement).toHaveAttribute(
+      "aria-label",
+      testString
+    );
+  });
+
+  test("should render with renderOpenComponent", async ({ mount, page }) => {
+    await mount(<WithRenderOpenButtonComponent />);
+
+    const buttonComponent = getComponent(page, "button");
+    await buttonComponent.getByText("Test");
```

Comment:
**suggestion**: I might be being dense and in need of a coffee - but what is this actually `awaiting`? It certainly doesn't modify what `buttonComponent` is referring to, which is the thing we're asserting against on the next line.

Assuming we want to ensure we're getting the button with the specific text "test", I'd just do that directly, with a locator like `const buttonComponent = page.getByRole("button", { name: "Test" });`, or even `const buttonComponent = page.getByText("Test")` if we're confident there's no other element with that text.

Reply:
I'll get it changed to `const buttonComponent = page.getByText("Test")` as that makes more sense. There are no other specific text using `Test` that could cause issues.